packages/swapper/src/swappers/PortalsSwapper/getPortalsTradeQuote/getPortalsTradeQuote.ts

```diff
@@ -86,124 +91,171 @@ export async function getPortalsTradeQuote(
+    sender: sendAddress,
+    inputToken,
+    outputToken,
+    inputAmount: sellAmountIncludingProtocolFeesCryptoBaseUnit,
+    slippageTolerancePercentage: userSlippageTolerancePercentageDecimalOrDefault,
+    partner: getTreasuryAddressFromChainId(sellAsset.chainId),
+    feePercentage: affiliateBpsPercentage,
+    validate: true,
+    swapperConfig,
+  })
+    .then(res => Ok(res))
+    .catch(async err => {
+      if (err instanceof PortalsError) {
+        // We assume a PortalsError was thrown because the slippage tolerance was too high during simulation
+        // So we attempt another (failing) call with autoslippage which will give us the actual expected slippage
+        const portalsExpectedSlippage = await fetchPortalsTradeOrder({
+          sender: sendAddress,
+          inputToken,
+          outputToken,
+          inputAmount: sellAmountIncludingProtocolFeesCryptoBaseUnit,
+          autoSlippage: true,
+          partner: getTreasuryAddressFromChainId(sellAsset.chainId),
+          feePercentage: affiliateBpsPercentage,
+          validate: true,
+          swapperConfig,
+        })
+          // This should never happen but could in very rare cases if original call failed on slippage slightly over 2.5% but this one succeeds on slightly under 2.5%
+          .then(res => res.context.slippageTolerancePercentage)
+          .catch(err => (err as PortalsError).message.match(/Expected slippage is (.*?)%/)![1])
+
+        // This should never happen as we don't have auto-slippage on for `/portal` as of now (2024-12-06, see https://github.com/shapeshift/web/pull/8293)
+        // But as soon as Portals implement auto-slippage for the estimate endpoint, we will most likely re-enable it, assuming it actually works
+        if (err.message.includes('Auto slippage exceeds'))
+          return Err(
+            makeSwapErrorRight({
+              message: err.message,
+              details: {
+                expectedSlippage: bn(portalsExpectedSlippage).toFixed(2, BigNumber.ROUND_HALF_UP),
+              },
+              cause: err,
+              code: TradeQuoteError.FinalQuoteMaxSlippageExceeded,
+            }),
+          )
+        if (err.message.includes('execution reverted'))
+          return Err(
+            makeSwapErrorRight({
+              message: err.message,
+              details: {
+                expectedSlippage: bn(portalsExpectedSlippage).toFixed(2, BigNumber.ROUND_HALF_UP),
+              },
+              cause: err,
+              code: TradeQuoteError.FinalQuoteExecutionReverted,
+            }),
+          )
+      }
       return Err(
         makeSwapErrorRight({
-          message: `unsupported ChainId`,
-          code: TradeQuoteError.UnsupportedChain,
-          details: { chainId: input.chainId },
+          message: 'failed to get Portals quote',
+          cause: err,
+          code: TradeQuoteError.NetworkFeeEstimationFailed,
         }),
```

Comment:
[comment]: Its times like this where i wish we had

```typescript
return SwapErr({ ... })
```

instead of

```typescript
return Err(makeSwapErrorRight({ ... }))
```

Reply:
soon.jpg?

Reply:
https://github.com/shapeshift/web/issues/8315